박스오피스 [STEP 1] John

limjae opened the pull request:
@zdodev 

첫 PR 올립니다! 😃



## 고민한점

### 1. 테스트

Decodable 객체로 파싱을 하기 전에 들어온 객체가 JSON 형태를 갖추고 있는지 확인하기 위해 다음과 같은 테스트를 더 추가했습니다.

```swift
func test_JSONSerialization테스트() {
        guard let data = data else {
            XCTAssertTrue(false)
            return
        }
        do {
            let parsed = try JSONSerialization.jsonObject(with: data, options: []) as? [String : Any]
            XCTAssertNotNil(parsed)
        } catch {
            print("error",error)
            XCTAssertTrue(false)
            return
        }
    }
```




## 궁금한점

현재 json 값은 숫자까지 전부 String 형태로 가지고 있고 이 데이터를 어떻게 사용 할 지 아직 몰라 우선 전부 String으로 저장했습니다. 추후 해당 데이터를 다른 타입으로 사용해야 한다면 Decodable 구조체의 property의 타입을 초기화 시점에 다른 타입으로 변환하여 저장하는 방법을 이용하려합니다.

이 때 json의 값이 String 형태로 들어 올 때 초기화 단계에서 바로 쓰고자 하는 형태로 변환하는 것이 효과적일까요? 아니면 다른 구조체를 선언하여 다른 구조체의 형태로 바꿔주는 함수를 추가하는 것이 나을 지 궁금합니다!

limjae committed:
chore: 파일 디렉토리 변경

limjae committed:
feat: 제공된 JSON 데이터 파싱 모델 추가

limjae committed:
refactor: 미사용 print 삭제

zdodev commented on BoxOfficeTests/BoxOfficeSearchResultTests.swift:
```diff
@@ -0,0 +1,46 @@
+
+import XCTest
+
+final class BoxOfficeSearchResultParsingTests: XCTestCase {
```
좋습니다.👍

zdodev commented on BoxOffice/Model/BoxOfficeSearchResult.swift:
```diff
@@ -0,0 +1,78 @@
+
+struct DailyBoxOffice: Decodable {
+    let audienceAccumulate: String
+    let audienceIncrement: String
```
`audienceCount` 는 어떨까요? 증가에 대한 의미보다 수에 대한 의미가 더 좋지 않을까 싶습니다!

limjae replied:
해당 사항 반영했습니다!
그리고 sales의 경우 Count의 의미를 내포하고 있으므로 생략했습니다.

zdodev commented on BoxOffice/Model/BoxOfficeSearchResult.swift:
```diff
@@ -0,0 +1,78 @@
+    let screenCount: String
+    let showCount: String
+
+    enum CodingKeys: String, CodingKey {
```
`CodingKey` 를 잘 활용해보셨네요.👍

zdodev reviewed (changes requested):
안녕하세요. @limjae !

Step 1 진행잘해주셨습니다.
모델 타입도 잘 정의해주셨습니다.👍

개인적으로 데이터를 전달받을 때, 원본의 타입을 변경하지 않고(ex: 서버 응답) 그대로 전달받은 다음 앱에서 필요한 형태로 가공하는 방식을 선호합니다!

코멘트 확인 부탁드립니다!

limjae committed:
refactor: 이름, 구조 변경

- result는 사용하지 않는 무의미한 구조라고 판단하여 Custom Decodable 구조체 선언
- DailyBoxOffice 구조체 역시 DailyBoxOfficeSearchResult내에서만 사용하므로 nested struct로 변경

zdodev commented on BoxOffice/Model/DailyBoxOfficeSearchResult.swift:
```diff
@@ -0,0 +1,84 @@
+    
+    struct DailyBoxOffice: Decodable {
+        let audienceAccumulate: String
+        let audienceCount: String
+        let audienceCountChange: String
+        let audienceCountChangeRate: String
```
👍

zdodev reviewed (approved):
안녕하세요. @limjae !

다음 스텝도 화이팅입니다!
고생하셨습니다!
